ignite/cmd/plugin_test.go

```diff
@@ -255,3 +275,141 @@ func dumpCmd(c *cobra.Command, w io.Writer, ntabs int) {
 		dumpCmd(cc, w, ntabs)
 	}
 }
+
+func TestLinkPluginHooks(t *testing.T) {
+	buildRootCmd := func() *cobra.Command {
+		var (
+			rootCmd = &cobra.Command{
+				Use: "ignite",
+				RunE: func(*cobra.Command, []string) error {
+					return nil
+				},
+			}
+			scaffoldCmd = &cobra.Command{
+				Use: "scaffold",
+				RunE: func(*cobra.Command, []string) error {
+					return nil
+				},
+			}
+			scaffoldChainCmd = &cobra.Command{
+				Use: "chain",
+				RunE: func(*cobra.Command, []string) error {
+					return nil
+				},
+			}
+		)
+		scaffoldCmd.AddCommand(scaffoldChainCmd)
+		rootCmd.AddCommand(scaffoldCmd)
+		return rootCmd
+	}
+	tests := []struct {
+		name            string
+		pluginInterface pluginInterface
+		shouldError     bool
```

Comment:
Let's check against `expectedError string` instead of a bool as in `TestLinkPluginsCmd` above

Then: 

```go
if tt.shouldError {
    // ...
}
```

becomes:

```go 
if tt.expectedError != "" {
    // ...
}

Reply:
This is not resolved, a field `expectedError` that contains the error label is better than just a `bool` because it allows to assert the kind of error.

```go

			linkPluginHooks(rootCmd, p)

			if tt.expectedError != "" {
				require.EqualError(p.Error, tt.expectedError)
				return
			}
			require.NoError(p.Error)
```

Reply:
did not mean to resolve this, has been updated

Reply:
removing it and adding `expectedArgs` to assert on arguments past to each phase of the hook life cycle

Reply:
sure, feel free to strengthen the test even more :)